ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx

```diff
@@ -241,6 +242,15 @@ export class ArchivedWorkflowDetails extends BasePage<RouteComponentProps<any>,
             </>
         );
     }
+                            
+    private cloneLabelsSafely() {
+        //we want to copy all labels except the ones that are added during workflow execution
+        const labelsArray = Object.keys(this.state.workflow.metadata.labels)
+            .map(k => ({ key: k, value: this.state.workflow.metadata.labels[k] }))
+            .filter(e => !e.key.startsWith("workflows.argoproj.io/") || e.key === "workflows.argoproj.io/workflow-template");
```

Comment:
I think this is the logic you want:

https://github.com/argoproj/argo-workflows/blob/ab786ecba6eb3e9d3fa7a717ded42727b8b64df8/workflow/util/util.go#L629